internal/setup/setup.go

```diff
@@ -0,0 +1,195 @@
+					Default: true,
+				},
+					&shouldClone,
+				)
+				if err != nil {
+					return fmt.Errorf("prompting for catalog cloning: %w", err)
+				}
+
+				// Prompt user for catalog repo
+				err = survey.AskOne(&survey.Input{
+					Message: "📦 What's your catalog repo URL?",
+				},
+					&catalogRepo,
+					survey.WithValidator(survey.Required),
+				)
+				if err != nil {
+					return fmt.Errorf("prompting for catalog repo: %w", err)
+				}
+			}
+
+			// Clone catalog
+			if shouldClone {
+				cmd := exec.Command("git", "clone", catalogRepo, catalogDir)
+				output, err := cmd.CombinedOutput()
+				if err != nil {
+					return fmt.Errorf("cloning catalog: %s", output)
+				}
+				fmt.Printf("✅ Cloned catalog from %s to %s\n", style.Link(catalogRepo), style.Code(catalogDir))
+			} else {
+				fmt.Println("😬 Sorry, cannot continue without catalog!")
+				os.Exit(1)
+			}
+
+		} else {
+			return fmt.Errorf("checking for catalog directory %s: %w", catalogDir, err)
+		}
+	}
+
+	// Check catalog directory
+	cat, err := catalog.Load(catalog.LoadOpts{
+		Dir:          catalogDir,
+		LoadEnvs:     true,
+		LoadProjects: true,
+		LoadReleases: true,
+		ResolveRefs:  true,
+	})
+	if err != nil {
+		fmt.Printf("🤯 Whoa! Found the catalog, but it's speaking gibberish. Check this error and try again:\n%v\n", err)
+		os.Exit(1)
+	}
+
+	// Print catalog content summary
+	envCount := len(cat.Environments)
+	projectCount := len(cat.Projects)
+	releaseCount := len(cat.Releases.Items)
+	if envCount == 0 || projectCount == 0 || releaseCount == 0 {
+		fmt.Print("🦗 Crickets... No ")
+		if envCount == 0 {
+			fmt.Print("environments")
+		}
+		if projectCount == 0 {
+			if envCount == 0 {
+				fmt.Print("/")
+			}
+			fmt.Print("projects")
+		}
+		if releaseCount == 0 {
+			if envCount == 0 || projectCount == 0 {
+				fmt.Print("/")
+			}
+			fmt.Print("releases")
+		}
+		fmt.Println(" found. Please add some to the catalog.")
+	} else {
+		fmt.Printf("🤩 Wowza! Your catalog's bursting with %d environments, %d projects, and %d releases.\n", envCount, projectCount, releaseCount)
+	}
+
+	// Try loading config file from given or default location
+	cfg, err := config.Load(configDir, catalogDir)
+	if err != nil {
+		return fmt.Errorf("loading config: %w", err)
+	}
+
+	// Save config file
+	cfg.CatalogDir = catalogDir
+	err = cfg.Save()
+	if err != nil {
+		return fmt.Errorf("saving config: %w", err)
+	}
+	fmt.Printf("✅ Saved config file to %s\n", style.Code(cfg.FilePath))
+
+	// Check for required and optional dependencies
+	fmt.Println(separator)
+	fmt.Print("🧐 Hmm, let's now see what dependencies you've got humming under the hood...\n\n")
+	missingRequired := false
+	for _, dep := range dependencies.AllRequired {
+		if dep.IsInstalled() {
+			fmt.Printf("✅ Found required %s dependency.\n", style.Code(dep.Command))
+		} else {
+			fmt.Printf("❌ Missing required %s dependency.\n", style.Code(dep.Command))
+			missingRequired = true
```

Comment:
We had that discussion in a Jam Session or other meeting (maybe you were not there, not sure), but we decided not to automate the installation of dependencies in `joy setup` for now.  If it was only for Mac and only with `brew`, it should not be too complicated, but it adds one more dependency (`brew`!) and making it cross-platform and working independently of brew (for those without it) is a whole different story.  Maybe we could create a ticket as a "Someday Maybe" to keep this on our radar?

Reply:
@j-martin I'm against a tool _automatically_ installing things for the user on their machine, specially through a required setup step. The tool can't know how an environment is set up—maybe the user has a custom version of tool compiled by themselves and I want to use that, or they're using nix for CLI stuff and brew just for casks and now they have to clean up brew packages that they didn't want, or they're specifically using `gcloud components` in order to have their Kubernetes tools automatically follow the correct versions as their work clusters and now joy introduced a subtle version incompatibility they'll have to debug later, etc.

People who'll be using joy will be power users, so these are likely scenarios. If we go for this feature in the future, my suggestion would be making it a prompt to install automatically for the user, or just output a "tooltip" with the command for them to run themselves.

Reply:
> The tool can't know how an environment is set up—maybe the user has a custom version of tool compiled by themselves and I want to use that, or they're using nix for CLI stuff and brew just for casks and now they have to clean up brew packages that they didn't want, or they're specifically using gcloud components in order to have their Kubernetes tools automatically follow the [correct versions](https://kubernetes.io/releases/version-skew-policy/#kubectl) as their work clusters and somehow joy didn't pick up on that and introduced a subtle version incompatibility they'll have to debug later, etc.

I would argue that if the user has their executable the way they like, `command -v` will find it, and nothing will be installed. 

My main concern is reducing friction as much possible and focusing on the developer experience. Having paired with a few people already on `joy`, I heard "why do I need to set up this and this up now, I just want to bump my release in staging". I had to defend `joy`, but once they got it setup they liked it. There are quite a few hoops to jump through to get `joy` working right now.

> People who'll be using joy will be power users

My understanding is that it will be used by all devs and, in some cases (?) QA for at least promoting their releases to staging. I doubt most dev care about having their specific version of `kubectl`, and if they do nothing will be installed. 

> prompt to install automatically for the user

I think that would be a totally reasonable thing to do and a decent compromise.

Reply:
> > prompt to install automatically for the user

> I think that would be a totally reasonable thing to do and a decent compromise.

Sure, that's a good compromise.

There's an alternative that I think achieves your goal of ease of installation to minimize friction and avoids automatic installation mechanisms in the binary:

In my opinion the best thing to do would be to not include this mechanism on the binary itself and instead leverage whatever packager manager we use to publish joy mechanism for dependencies. All of the mainstream ones have an "optional" dependency mechanism.

In the case of brew, we list the things that are required for joy to work as dependencies, and any extra tools [as "optional" or "recommended"](https://docs.brew.sh/Formula-Cookbook#specifying-other-formulae-as-dependencies). If the user attempts to use a command that is optional and requires an optional dependency, we put a little bit of logic to output a message `Please run brew install --include-optional` or `--with-optional foo` or whatever.

This achieves: `brew install --include-optional joy` installs everything necessary in one command, no need for the `joy setup` prompt, and power users can use other mechanisms like `brew install --skip-recommended` to not install something like kubectl if they already have a specific one that they need or any other power user mechanisms from brew/apt/yum/build from source/PATH magic that they like. Bonus we don't add any extra logic in the binary for doing these useful things.

Reply:
> My understanding is that it will be used by all devs and, in some cases (?) QA for at least promoting their releases to staging. I doubt most dev care about having their specific version of kubectl, and if they do nothing will be installed.

Small comment because I missed this the first time around: agreed, but joy is getting published as an open source project as well, so some minimal consideration for external use cases is good for it. But I do agree that the main consideration should be to optimize it for our use case.

Reply:
I think that _especially_ for `kubectl` we should not try to install it for user, as it's useless unless properly configured. As I mentioned before, we don't merely need the binary to be available, kube contexts also need to be configured properly.  And, foremost, it's only needed by `joy sealed-secret import`, which is very power-user and mostly intended for devops/platform people to extract sealed-secret certs from K8s to joy `Environments`.

To use @riclima's expression, I think we're really bike-shedding this and losing track of the big picture.  I think we all agree that making for a rather simple and streamlined on-boarding experience is important, but I think we should draw the line at some point, especially on the short term.  We will already have a very pleasing first `joy setup` experience, which may evolve eventually, but that is only relevant on first-time installs, there are so many other core features we should be implementing or improving as soon as possible...